### PR TITLE
NEO-2312: Fixing Table scenarios and adding new story

### PR DESCRIPTION
**Pull request by enrique-prado**
[IconButton inside Table Cell](https://deploy-preview-462--neo-react-library-storybook.netlify.app/?path=/story/components-table--with-icon-button)

[Standalone IconButton with Badge](https://deploy-preview-462--neo-react-library-storybook.netlify.app/?path=/story/components-iconbutton--with-badge)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged `@avaya-dux/dux-devs`

This PR fixes rendering issues when a ButtonIcon component is inside a Neo Table cell. 

`@avaya-dux/dux-design`: Nothing new to review since you last reviewed it.

`@avaya-dux/dux-devs`: css code has increased specificity to fix Table scenarios. Also added new Table story to showcase scenario.

- IconButton inside Table cell should render properly
- Stand alone IconButtons should also render as expected.

image

## Summary by CodeRabbit

- **New Features**
  - Introduced an "Add Note" icon button within the table component to enhance interactivity.
  
- **Style**
  - Improved styling and positioning of badges and icons within table cells for a more polished appearance.

- **Chores**
  - Updated the `@avaya/neo-react` package version from `1.1.32` to `1.1.33`.

### CODE REVIEW

File: src/components/Table/Table.stories.tsx [hunk 1]
```diff
@@ -101,6 +102,29 @@ export const MoreActionsMenu = () => (
...
 
+export const WithIconButton = () => (
+	<Table
+		caption="Last column has an IconButton."
+		columns={[
+			...FilledFields.columns,
+			{
+				Header: "Add Note",
+				Cell: () => (
+					<IconButton
```
Comment (enrique-prado) [inline src/components/Table/Table.stories.tsx:113]: Adding IconButton to the last column.

File: src/components/IconButton/IconButton_shim.css [hunk 2]
```diff
@@ -1,8 +1,19 @@
-.neo-btn.neo-icon-btn {
+.neo-btn.neo-badge::after {
+	right: 5px;
+	top: 5px;
+}
+
+.neo-table td .neo-btn.neo-btn-square,
```
Comment (joe-s-avaya) [inline src/components/IconButton/IconButton_shim.css:6]: this line doesn't seem necessary, is it?

**Review by joe-s-avaya — CHANGES_REQUESTED** [review]
you seem to have duplicate and unused classes in your CSS

File: src/components/IconButton/IconButton_shim.css (hunk 2 above)
Comment (enrique-prado) [reply src/components/IconButton/IconButton_shim.css:6]: It is, this is when the row has focus / selected. Without it the icon's display position is set to block and it looks wrong.

File: src/components/IconButton/IconButton_shim.css [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
-.neo-btn.neo-badge::after {
-	right: 7px;
-	top: 4px;
+.neo-table td span[class*=neo-icon-state--large]::before {
+	font-size: 32px;
+}
+
+.neo-table td .neo-btn.neo-badge::after {
+	right: 6px;
+	top: 11px;
```
[the remark this answers is not in the document]
Comment (enrique-prado) [reply src/components/IconButton/IconButton_shim.css:18]: I spent a bunch of time trying to figure out why the badge is position differently when the IconButton is inside the table vs when it is standalone but was unable to find the reason, therefore I had to play with the offsets for each scenario to make them look the same.

**Pull request merged**